presto-main/src/main/java/com/facebook/presto/operator/PagesIndex.java

```diff
@@ -87,6 +87,44 @@ public PagesIndex(List<Type> types, int expectedPositions)
         }
     }
 
+    public interface Factory
+    {
+        PagesIndex newPagesIndex(List<Type> types, int expectedPositions);
+    }
+
+    public static class TestFactory
```

Comment:
The convention is to use the prefix `Testing` for classes used for testing, so name this `TestingFactory`. A `Test` prefix is for classes containing unit tests.